src/main/java/com/scrive/pdftools/tools/flattener/LambdaFlattener.java

```diff
@@ -0,0 +1,41 @@
+package com.scrive.pdftools.tools.flattener;
+
+import com.itextpdf.text.DocumentException;
+import com.itextpdf.text.exceptions.BadPasswordException;
+import com.itextpdf.text.pdf.PdfReader;
+import com.itextpdf.text.pdf.PdfStamper;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class LambdaFlattener {
+
+    public static ByteArrayOutputStream execute(FlattenerSpec flatSpec) throws IOException, DocumentException {
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        try (ByteArrayInputStream file = new ByteArrayInputStream(flatSpec.getFileContent())) {
+            PdfReader reader = new PdfReader(file);
+            try(PdfStamperAutoclosable stamper = new PdfStamperAutoclosable(reader, bos)) {
+                stamper.setFormFlattening(true);
+                stamper.setAnnotationFlattening(true);
+            } catch (BadPasswordException e) {
+                PdfReader.unethicalreading = true;
+                try(PdfStamperAutoclosable stamper = new PdfStamperAutoclosable(reader, bos)) {
+                    stamper.setFormFlattening(true);
+                    stamper.setAnnotationFlattening(true);
```

Comment:
@GustafNi if you are referring to the fact that we twice instantiate the stamper (at the cost of additional try-with-resources) and that we duplicate the `setFormFlattening(); setAnnotationFlattening()` logic we can extract the common parts into a method and simply call it twice:
```suggestion
            try {
                flatten(reader, bos);
            } catch (BadPasswordException e) {
                PdfReader.unethicalreading = true;
                flatten(reader, bos);
```

with 
```java
private static flatten(reader, bos) throws ... {
     try (PdfStamperAutoclosable stamper = new PdfStamperAutoclosable(reader, bos)) {
           stamper.setFormFlattening(true);
           stamper.setAnnotationFlattening(true);
      }
}

Reply:
Cool. Yes i was mostly wondering if there was something else I didn't understand about this try-with-resources statement. It's all clear now. Yep I created a function that performs the flattening now.